src/common/plugin.ts

```diff
@@ -45,15 +48,38 @@ export abstract class Plugin implements IPlugin {
     const channelName = channel.name;
     const categoryName = channel.parent?.name.toLowerCase();
 
-    if (this.pluginChannelName && this.pluginChannelName !== channelName) {
+    if (this.pluginCategoryName) {
+      // If category matches, move on
+      if (this.pluginCategoryName.toLowerCase() !== categoryName) {
+        // If channel name matches, move on
+        if (this.pluginChannelNames) {
+          if (!this.pluginChannelNames.includes(channelName)) {
```

Comment:
Why is this so complicated

Reply:
As for some constructive feedback, this code could simplify some of a lot its branches by using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

Reply:
I'm also disappointed in how complicated this turned out; however, I couldn't figure out how to avoid it. I couldn't simply combine the null checks and equality checks, as it's necessary to know whether each field is defined, as well as whether the current channel matches it if it is. Without that granularity, there's no way I could see to prevent certain checks from occurring when they shouldn't, such as checking whether the channel name matched when the category name was already determined to match.
The short explanation for the complexity of this code is that there are 5 possible outputs depending on which restrictions the developer places on the plugin.
A solution I can think of is to scrap the optional list of channel names, or otherwise get rid of the old single channel name in favor of it (which would involve lots of refactoring).